src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -326,10 +326,17 @@ class BrainpoolP512R1(EllipticCurve):
 
 
 class ECDSA(EllipticCurveSignatureAlgorithm):
-    def __init__(self, algorithm):
+    def __init__(
+        self,
+        algorithm: typing.Union[asym_utils.Prehashed, hashes.HashAlgorithm],
+    ):
         self._algorithm = algorithm
 
-    algorithm = utils.read_only_property("_algorithm")
+    @property
```

Comment:
Is type information lost here if you just do `algorithm = property(lambda self: self._algorithm)`?

Reply:
`mypy` complained that it didn't match the parent declaration, yeah.

Reply:
Could we get a comment here explaining the need for this extra boilerplate and pointing at either an existing or new mypy bug?

Reply:
It's not a mypy bug -- this implements an interface and so the property needs to match teh types on that interface.

It's a bug of `read_only_property` that it goes to `Any` and thus disables the type checking.

I don't see a need for a comment on what amounts to a straightforward function.

Reply:
Are you saying that using `property` as a non-decorator also always returns `Any` rather than the underlying type of the property? I understand that `read_only_property` did that since we wrapped the property declaration ourselves and therefore it implicitly returned `Any`. That's unfortunate.

Reply:
I'm saying that `read_only_property` was a whole `Any` thing. This effectively disabled mypy.

Then `property(lambda self: self.foo)` had no type signature so `mypy` was like "hey, your parent class has a type signature so you need one too".

Reply:
Ah, I see. It's a shame there's no way to wrap a value to be read only and proxy the type information. Oh well.